### PR TITLE
[11.0] FWD port search form improvements

### PR DESCRIPTION
**Pull request by simahawk**
From #66

### CODE REVIEW

File: cms_form/models/cms_search_form.py [hunk 1]
```diff
@@ -128,15 +140,24 @@ def form_search_domain(self, search_values):
                 if not value:
                     continue
                 operator = 'in'
-            elif field['type'] in ('many2one', ) and not value:
-                # we need an existing ID here ( > 0)
-                continue
+            elif field['type'] in ('many2one', ):
+                value = int(value) if value.isdigit() else 0
```
Comment (ghost) [inline cms_form/models/cms_search_form.py:144]: any chance to run into smth w/ spaces here? I.e., `'3 '.isdigit()` is `False`

File: cms_form/models/cms_search_form.py [hunk 2]
```diff
@@ -24,6 +24,11 @@ class CMSFormSearch(models.AbstractModel):
     _form_results_orderby = ''
     # declare fields that must be searched w/ multiple values
     _form_search_fields_multi = ()
+    # declare custom domain computation rules
+    _form_search_domain_rules = {
+        # field name: (leaf field name, operator, format value),
+        # 'product_id': ('product_id.name', 'ilike', '{}'),
```
Comment (ghost) [inline cms_form/models/cms_search_form.py:30]: is that a usage example? :slightly_smiling_face:

**Review by tschanzt — APPROVED** [review]

File: cms_form/models/cms_search_form.py (hunk 1 above)
Comment (simahawk) [reply cms_form/models/cms_search_form.py:144]: that's impossible, at this stage the value is clean ;)

File: cms_form/models/cms_search_form.py (hunk 2 above)
Comment (simahawk) [reply cms_form/models/cms_search_form.py:30]: yes, don't complain! :D

**Pull request merged**